internal/interface/handlers/middleware_handler.go

```diff
@@ -56,3 +58,51 @@ func (h *MiddlewareHandler) IsLogin(c *fiber.Ctx) error {
 	// move to next handlers
 	return c.Next()
 }
+
+func (h *MiddlewareHandler) SGCUAdmin(c *fiber.Ctx) error {
+	h.IsLogin(c)
+
+	userDTO := c.Locals("user").(*dtos.UserDTO)
+	if userDTO.Role != constant.SGCU_ADMIN && userDTO.Role != constant.SGCU_SUPERADMIN {
+		resp := response.NewResponseFactory(response.ERROR, errors.New("Forbidden").Error())
+		return resp.SendResponse(c, fiber.StatusForbidden)
+	}
+
+	return c.Next()
+}
```

Comment:
if we use this middleware, no need to put IsLogin() into fiber handler right ?

ex. 
if we want to allow "sgcu_admin" and "sgcu_superadmin" only access route, we just use SGCUAdmin middleware

in the same time if we just want to allow only "sgcu_superadmin" to access just use SGCUSuperAdmin

Reply:
yes, let me try it first tho555

Reply:
If so could you fix user router dai maii 🙇🏻‍♂️ @bookpanda

Reply:
or you want the another way kub @bookpanda,
just dont call IsLogin() inside each middleware

Reply:
ok